src/debounce/mod.rs

```diff
@@ -507,3 +513,31 @@ fn restart_timer(timer_id: &mut Option<u64>, path: PathBuf, timer: &mut WatchTim
     }
     *timer_id = Some(timer.schedule(path));
 }
+
+fn handle_on_going_write_event(timer: &WatchTimer, path: PathBuf, tx: &mpsc::Sender<DebouncedEvent>) {
+    let mut on_going_write_event = timer.on_going_write_event.lock().unwrap();
+    let mut emitted = false;
+    let mut to_be_scheduled = false;
+    if let Some(ref i) = *on_going_write_event {
+        let now = Instant::now();
+        if i.0 <= now {
+            //fire event
+            let _ = tx.send(DebouncedEvent::OnGoingWrite((i.1).clone()));
+            emitted = true;
+        }
+    } else {
+        //schedule event
+        if let Some(_) = timer.on_going_write_duration {
+            to_be_scheduled = true;
+        }
+    }
+
+    if to_be_scheduled {
+        let duration = timer.on_going_write_duration.unwrap();
+        let tt = Instant::now() + duration;
+        *on_going_write_event = Some((tt, path));
+    }
+    if emitted {
+        *on_going_write_event = None;
+    }
```

Comment:
These two clauses should be moved inline to where their condition is set... unless I've missed something, there's no advantage to having them down there. They'll never both run, and there's no other place where their conditions get switched on.

Reply:
First of all I apologize for this blunder. I should have called out this code. The problem is `on_going_write_event` is borrowed when I `*on_going_write_event` it on line 520. So I cannot call `*on_going_write_event` again (to set it to `None` or `Some` based on if condition) because I get `cannot borrow as mutable more than once` error. 

I have made this better but still not inline. Let me know what you think. I'll keep looking for better options.

Reply:
That looks good :) :+1: